### PR TITLE
`gtest`: run tests in parallel

### PR DESCRIPTION
**Pull request by ukint-vs**
- run tests and fixtires in parallel
- one time run instead of new run with each expected

todo:
- [x] Fix runtests cli
- [x] Print logs only if error

@gear-tech/dev

### CODE REVIEW

**Comment by NikVolf** [comment]
how about log processing ?

**Comment by ukint-vs** [comment]
> how about log processing ?

log will be reworked in a different PR

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -125,4 +125,4 @@ jobs:
         run: ./scripts/gear.sh build examples-proc
 
       - name: "Test: Process gear-test"
-        run: ./scripts/gear.sh test gtest -vv
\ No newline at end of file
+        run: ./scripts/gear.sh test gtest
```
Comment (breathx) [inline .github/workflows/CI.yaml:128]: Would be good add newline at the end of file

File: gtest/src/check.rs [hunk 2]
```diff
@@ -20,16 +20,78 @@ use crate::js::{MetaData, MetaType};
...
+
+impl Log for FixtureLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        self.inner.enabled(metadata)
+    }
+
+    fn log(&self, record: &Record) {
+        // Check if the record is matched by the logger before logging
+        if self.inner.matches(record) {
+            let mut map = self.map.write().expect("RwLock poisoned");
```
Comment (breathx) [inline gtest/src/check.rs:84]: Doesn't it rough panic if locker is poisoned?

File: gtest/src/check.rs [hunk 3]
```diff
@@ -20,16 +20,78 @@ use crate::js::{MetaData, MetaType};
...
+
+            map.entry(thread::current().id())
+                .or_default()
+                .push(record.args().to_string());
+        }
+    }
+
+    fn flush(&self) {}
```
Comment (breathx) [inline gtest/src/check.rs:92]: Don't we need flushing logic?

File: gtest/src/runner.rs [hunk 4]
```diff
@@ -272,24 +284,36 @@ where
...
+            let do_run = step_no < messages.len();
             if step_no < messages.len() {
                 let mut run_result = runner.run_next(messages[step_no].clone(), u64::MAX);
                 runner.process_wait_list(&mut run_result);
 
                 log::info!("step: {}", step_no + 1);
 
                 if run_result.any_traps() && step_no + 1 == steps {
-                    result = Err(anyhow::anyhow!("Runner resulted in a trap"));
+                    _result = Err(anyhow::anyhow!("Runner resulted in a trap"));
```
Comment (breathx) [inline gtest/src/runner.rs:295]: Unused var?

**Review by breathx — APPROVED** [review]
Current changes has some minor issues and refactoring proposals, but they will be done in the closest future.

Anyway, this strongly improves our testing system. 
Good job!

**Pull request merged**